### ISSUE TITLE
Would this be a good way to train?

### ISSUE THREAD

**Issue by ohgyeongsu** [ultralytics/ultralytics#21740]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

For example, I want to train a defect detection model based on image data with a size of 5000x5000.
Since the image is too large, I plan to first resize it to half the size, and then further split it into smaller patches for training.

In this case, would it be better to train only on the cropped images that contain objects, or should I also include background-only patches without objects in the training data?

Also, in my previous training, I implemented a `CustomTrainer` to apply custom cropping. If I want to train using cropped image patches instead, which part of the code should I modify?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ohgyeongsu, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will review your issue and assist you shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
- Your dataset details (image sizes, example images with labels, how patches are created)📸
- Training logs and exact commands or scripts used (Python/CLI)
- Any custom code snippets (e.g., CustomTrainer subclass and where cropping is applied)
- Clear objectives and evaluation metrics you’re targeting
- Hardware, OS, Python, PyTorch, and ultralytics versions
Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by RizwanMunawar** [comment]
@ohgyeongsu Hi! To help your model generalize better in complex scenarios, it’s important to include background information. Training only on cropped images can also lead to overfitting. If your goal is to improve model accuracy, the best approach is to use data augmentation. You can check out our detailed guide here: https://docs.ultralytics.com/guides/yolo-data-augmentation/. Please let me know if you have any questions. Thanks

**Comment by ohgyeongsu** [comment]
@RizwanMunawar 
When training an Ultralytics YOLO model, what part of the code do I need to customize to train on image patches?

**Comment by yyq19990828** [comment]
I think make a patched dataset is a straight way.

**Comment by glenn-jocher** [comment]
Agree—creating a patched dataset is the straightforward path. You don’t need to modify the Trainer; tile your 5000×5000 images offline, rewrite labels for each tile, then point your data.yaml to the patched folders and train normally. As a template, see the split-and-crop utility in split_dota (handles saving background-only tiles via allow_background_images): see the Reference for split_dota; adapt it to your label format. Example:

from ultralytics.data.split_dota import split_images_and_labels
split_images_and_labels(data_root="path/to/data_root", save_dir="path/to/tiles", split="train", crop_sizes=(1024,), gaps=(200,))

yolo detect train data=tiles.yaml model=yolo11n.pt imgsz=1024

See the split_dota docs for details: https://docs.ultralytics.com/reference/data/split_dota/
